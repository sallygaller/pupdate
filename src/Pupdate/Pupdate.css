.Pupdate {
    width: 50%;
    margin: 0px auto;
    display: flex;
    flex-direction: row;
    align-content: flex-start;
    justify-content: center;
    border-bottom: 1px solid black;
}

.Pupdate h3 {
    margin: 0px;
    padding: 10px;
}
.Pupdate button {
    margin: 0px;
    padding: 10px;
}

/* .Pupdate-item {
    padding: 15px;
    border: 1px solid black;
    flex-basis: content;
} */

.Pupdate-item {
    flex-basis: auto;
   padding: 10px;
}