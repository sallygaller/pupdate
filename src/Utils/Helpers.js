export default function playstyles(playstyle) {
  if (playstyle === "nervous") {
    return "I'm nervous or shy around other dogs.";
  }
  if (playstyle === "gentle") {
    return "I play gently.";
  }
  if (playstyle === "food-obsessed") {
    return "I'm food-obsessed!";
  }
  if (playstyle === "walks") {
    return "I like going on walks with my pup pals.";
  }
  if (playstyle === "parks") {
    return "I like going to dog parks with my pup pals.";
  }
  if (playstyle === "wrestling") {
    return "I like playfighting and wrestling.";
  }
  if (playstyle === "rambunctious") {
    return "I'm rambunctious and playful.";
  }
  if (playstyle === "ball-obsessed") {
    return "I love to play fetch!";
  }
}
