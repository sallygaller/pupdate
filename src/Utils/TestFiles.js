export const pupdates = [
  {
    id: 1,
    date: "2020-03-02T08:00:00.000Z",
    starttime: "09:30:00",
    endtime: "10:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park!",
    organizer: 1,
  },
  {
    id: 2,
    date: "2020-03-01T08:00:00.000Z",
    starttime: "15:30:00",
    endtime: "17:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park! I'll bring a frisbee",
    organizer: 1,
  },
];

export const pups = [
  {
    id: 1,
    name: "Eddie",
    breed: "dachshund",
    mix: true,
    age: "Adult",
    size: "M",
    nervous: true,
    rambunctious: false,
    gentle: true,
    wrestling: false,
    walks: true,
    parks: false,
    foodobsessed: true,
    ballobsessed: false,
    description:
      "Eddie can be a bit shy around other dogs at first, but he is really sweet! He loves going on long walks where there is a lot to sniff",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 1,
  },
  {
    id: 2,
    name: "Buddy",
    breed: "beagle",
    mix: true,
    age: "Adult",
    size: "L",
    nervous: false,
    rambunctious: true,
    gentle: false,
    wrestling: true,
    walks: false,
    parks: true,
    foodobsessed: false,
    ballobsessed: false,
    description:
      "Buddy can be a bit full on for some dogs, but he really just loves to play! He especially loves puppies",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 2,
  },
];
