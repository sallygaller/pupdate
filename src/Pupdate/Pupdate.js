import React from "react";
import { Link } from "react-router-dom";
import TokenService from "../services/token-service";
import { API_ENDPOINT } from "../config";
import moment from "moment";
import "./Pupdate.css";

class Pupdate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      organizerPups: [],
      attendees: [],
      error: null,
    };
  }

  componentDidMount() {
    // get pup of each pupdate organizer
    fetch(API_ENDPOINT + `/pups/user/${this.props.pupdate.organizer}`, {
      headers: {
        authorization: `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => res.json())
      .then((responseData) =>
        this.setState({
          organizerPups: responseData,
        })
      )
      .catch((error) => {
        console.error({ error });
      });
  }

  render() {
    const pupdate = this.props.pupdate;
    return (
      <div className="Pupdate">
        <div className="Pupdate-group">
          <div className="Pupdate-item">
            <p>
              Date: {moment(pupdate.date).format("LL")} <br></br>
              Time: {moment(pupdate.starttime, "h:mm A").format("h:mm A")} -
              {moment(pupdate.endtime, "h:mm A").format("h:mm A")}
            </p>
          </div>
          <div className="Pupdate-item">
            <p>Location: {pupdate.location}</p>
          </div>
          <div className="Pupdate-item">
            <ul>
              {this.state.organizerPups.map((organizerPup) => (
                <li key={organizerPup.id}>
                  {organizerPup.name}
                  <br></br>
                  <Link to={`/pups/${organizerPup.id}`}>
                    <button className="Pupdate-play-profile" type="button">
                      View Play Profile
                    </button>
                  </Link>
                </li>
              ))}
            </ul>
          </div>
          <div className="Pupdate-item">
            <Link to={`/pupdates/${pupdate.id}`}>
              <button className="Pupdate-rsvp" type="button">
                View Pupdate/Change RSVP
              </button>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Pupdate;
