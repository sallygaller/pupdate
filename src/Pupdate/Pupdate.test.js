import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Pupdate from "./Pupdate";
import { pupdates } from "../Utils/TestFiles";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Pupdate pupdate={pupdates[0]} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Pupdate pupdate={pupdates[0]} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
