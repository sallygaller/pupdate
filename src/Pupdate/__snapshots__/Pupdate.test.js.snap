// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Pupdate"
>
  <div
    className="Pupdate-group"
  >
    <div
      className="Pupdate-item-double"
    >
      <p>
        Date: 
        March 2, 2020
         
        <br />
        Time: 
        9:30 AM
         - 
        10:30 AM
         
        <br />
        Location: 
        Arbor Lodge Park
      </p>
    </div>
    <div
      className="Pupdate-item"
    >
      <ul>
        Organized by:
      </ul>
    </div>
  </div>
</div>
`;
